src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.cs

```diff
@@ -477,12 +482,21 @@ private static void OnGenericPropertyChanged(object dependencyObject, Dependency
 			}
 		}
 
+		protected override Foundation.Size MeasureOverride(Foundation.Size availableSize)
+		{
+			ViewportMeasureSize = availableSize;
+
+			return base.MeasureOverride(availableSize);
```

Comment:
it's possible here that ViewportMeasureSize is different than base.MeasureOverride(availablesize), is it expected?

Reply:
Yes, it's correct. The ItemsStackPanel needs to know the availableSize available to the ScrollViewer to know how many items it will want to materialize. (The panel itself will be measured with infinite height because it's inside the ScrollViewer.)

I'll add some green to make it a bit more clear.

Reply:
Done.